toolchain/check/testdata/interop/cpp/function/overloads.carbon

```diff
@@ -175,6 +175,44 @@ fn F() {
   Cpp.foo(-1);
 }
 
+// --- floating_point_literal.h
+
+auto foo(float a) -> void;
+auto foo(double a) -> void;
+
+// --- import_floating_point_literal.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "floating_point_literal.h";
+
+fn F() {
+  Cpp.foo(1.0);
+}
```

Comment:
```suggestion
// --- floating_point_literal.h

auto foo(float a) -> float*;
auto foo(double a) -> double*;

// --- import_floating_point_literal.carbon

library "[[@TEST_NAME]]";

import Cpp library "floating_point_literal.h";

fn F() {
  let d: f64* = Cpp.foo(1.0);
}
```
Consider something like this to test that we select the `double` overload here. (Another option might be to `= delete` the float overload.)